clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -787,11 +794,13 @@ public void close() {
         public final FutureRecordMetadata future;
         public final boolean batchIsFull;
         public final boolean newBatchCreated;
+        public final boolean retryForNewBatch;
```

Comment:
Maybe we can just call this `aborted` and set `batchIsFull` to indicate that the append was aborted because the batch was full?

Reply:
The only thing that makes me hesitate about batchIsFull is that technically the batch is empty. That is why we returned and needed to create a new batch. Would the aborted bool be enough to signal changing the partition?